Document Rancher image vulnerability scanning (#3364)
rancher/docs: content/rancher/v2.6/en/security/cve-scans/_index.md

---
title: Rancher Image Vulnerability Scanning
weight: 5
---

Rancher shares security scan results for the images in a Rancher release. This information includes CVE (Common Vulnerabilities and Exposures) IDs, status, notes or remediation plans where available.

The CVE scanning process, introduced in Rancher v2.6, reduces the patching cycle time of Rancher images.

- [Scope](#scope)
- [Scanning Tool](#scanning-tool)
- [Reporting Scan Results](#reporting-scan-results)
- [Addressing Vulnerabilities](#addressing-vulnerabilities)

# Scope

> The CVE scan includes all images shipped with Rancher releases, which are listed in the `images.txt` file included with every release. This list includes all core Rancher components and features.

Reviewer: Could mentioned here that the image-sources.txt also includes that images along with an annotation indicating what rancher feature uses the image.

Author: changed to:

> The CVE scan includes all images shipped with Rancher releases, which are listed in the `rancher-images.txt` file included with every release. This list includes all core Rancher components and features.
> 
> The `rancher-images-sources.txt` file also includes the images, along with an annotation indicating what Rancher feature uses the image.

# Scanning Tool

Rancher images are scanned with [Trivy,](https://github.com/aquasecurity/trivy) tool from Aqua Security. Other image scanning tools may yield different results.

# Scan Results

> The CVE scan report lists any image containing a CVE that is ranked by our scanning tool with a rating of `high` or above.

Reviewer: High is always seen fully capitalized so can change for consistency. To clarify, only one rating is above high, critical, so you could just say HIGH and CRITICAL.
```suggestion
The CVE scan report lists any image containing a CVE that is ranked by our scanning tool with a rating of `HIGH` or above.
```

> The report is provided in CSV format with each Rancher release. It is also available on a publicly accessible web page. This web page is updated weekly.

Reviewer: It's daily now. Should also include that the CSV info included on release is from the day of release and is likely to become outdated as new CVE's for an image version are detected.
```suggestion
The report is provided in CSV format with each Rancher release. It is also available on a publicly accessible web page. This web page is updated daily.
```

For each image listed in the scan report, the following information is listed:

- Image name
- CVE ID
- Severity level
- Package name
- Status of patch
  - For mirrored images, the current upstream state is used

>   - For Rancher images, the status in the Rancher pipeline is used

Reviewer: There is other info as well but it is subject to change so I think this covers ones that are likely to stay.

# Addressing Vulnerabilities

> As new CVEs are reported, or new images with high-level CVEs are brought into the pipeline, the Rancher team is automatically alerted and triages each vulnerability.

Reviewer: ```suggestion
As new CVEs are reported, or new images with high or critical severity CVEs are brought into the pipeline, the Rancher team is automatically alerted and triages each vulnerability.
```

Rancher addresses vulnerabilities differently depending on whether the image is mirrored from a community project or maintained by Rancher.

Some Rancher features will not receive security updates, or will receive limited ones. For details, see [this section](#features-with-limited-security-updates)

### Images Mirrored from Community Projects

If the upstream project has addressed and released a fix, the images are upgraded when possible in future Rancher releases.

If the upstream project has not released a fix, an issue explaining the situation is created and tracked by Rancher's engineering team.

### Images Maintained by Rancher

CVEs for images maintained by Rancher fall into one of the following categories:

- **False-positive:** We document these results and provide an explanation.
- **Vulnerabilities that will not be fixed:** We document these results and provide an explanation.For example, we may decide not to fix vulnerabilities caused by an upstream project that is no longer maintained, or a vulnerability that has a low attack surface.

> - **Vulnerabilities that can be fixed:** These results include vulnerabilities in Rancher projects or upstream packages that need to be updated, and that are within our ability to fix. They are addressed an released in the same way as any other issue within Rancher.

Reviewer: @cbron do we want to state our process like this? I feel like this may be interpreted as some sort of guarantee when in reality is just loosely the process we are following internally atm.

Author: condensed per Caleb's note https://rancher.slack.com/archives/C01DK1CKSUB/p1625176711048200

If the upstream project has not released a fix, an issue explaining the situation is created and tracked by Rancher's engineering team.

### Images Maintained by Rancher

CVEs for images maintained by Rancher fall into one of the following categories:

- **False-positive:** We document these results and provide an explanation.
- **Vulnerabilities that will not be fixed:** We document these results and provide an explanation.For example, we may decide not to fix vulnerabilities caused by an upstream project that is no longer maintained, or a vulnerability that has a low attack surface.
- **Vulnerabilities that can be fixed:** These results include vulnerabilities in Rancher projects or upstream packages that need to be updated, and that are within our ability to fix. They are addressed an released in the same way as any other issue within Rancher.

### Features with Limited Security Updates

The following features will receive limited or no security updates:

- Istio v1 (1.5.920) should no longer be used. We recommend using Istio v2. The images for Istio 1.5.920 should not be downloaded unless upgrading Istio is not feasible.

> - Pipelines were deprecated as of Rancher v2.5 and should no longer be used. We recommend handling Git-based deployment pipelines with  [Fleet.]({{<baseurl>}}/rancher/v2.6/en/deploy-across-clusters/fleet)

Reviewer 2: Pipelines needs note that it is getting critical CVE's fixes only

Author: updated
